### ISSUE TITLE
Cannot find module 'next/dist/compiled/regexr-lexer/lexer' on Azure server

### ISSUE THREAD

**Issue by rapritchard** [vercel/next.js#26945]
### What version of Next.js are you using?

10.2.3

### What version of Node.js are you using?

14.16.0

### What browser are you using?

Chrome, Firefox

### What operating system are you using?

Windows

### How are you deploying your application?

Azure with custom server

### Describe the Bug

My server used to work fine, however as of yesterday when trying to start up the next server on azure, it errors with what's below:
```bash
C:\home\site\wwwroot>npm run start

> nextjscheckout@0.1.0 start C:\home\site\wwwroot
> node server.js

started server on host localhost, port 3000, env production
info  - Loaded env from C:\home\site\wwwroot\.env.local
(node:13040) UnhandledPromiseRejectionWarning: Error: Cannot find module 'next/dist/compiled/regexr-lexer/lexer'
Require stack:
- C:\home\site\wwwroot\node_modules\next\dist\lib\load-custom-routes.js
- C:\home\site\wwwroot\node_modules\next\dist\next-server\server\next-server.js
- C:\home\site\wwwroot\node_modules\next\dist\server\next.js
- C:\home\site\wwwroot\server.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (C:\home\site\wwwroot\node_modules\next\dist\lib\load-custom-routes.js:1:496)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:13040) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:13040) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
info  - Using webpack 4. Reason: custom webpack configuration in next.config.js https://nextjs.org/docs/messages/webpack5
Error: Cannot find module 'next/dist/compiled/regexr-lexer/lexer'
Require stack:
- C:\home\site\wwwroot\node_modules\next\dist\lib\load-custom-routes.js
- C:\home\site\wwwroot\node_modules\next\dist\next-server\server\next-server.js
- C:\home\site\wwwroot\node_modules\next\dist\server\next.js
- C:\home\site\wwwroot\server.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.mod._resolveFilename (C:\home\site\wwwroot\node_modules\next\dist\build\webpack\require-hook.js:4:1784)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (C:\home\site\wwwroot\node_modules\next\dist\lib\load-custom-routes.js:1:496)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'C:\\home\\site\\wwwroot\\node_modules\\next\\dist\\lib\\load-custom-routes.js',
    'C:\\home\\site\\wwwroot\\node_modules\\next\\dist\\next-server\\server\\next-server.js',
    'C:\\home\\site\\wwwroot\\node_modules\\next\\dist\\server\\next.js',
    'C:\\home\\site\\wwwroot\\server.js'
  ]
}

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! nextjscheckout@0.1.0 start: `node server.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the nextjscheckout@0.1.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\local\AppData\npm-cache\_logs\2021-07-06T08_27_34_482Z-debug.log
```

Checking the node_modules on my local machine and the server shows that the module its looking for does not exist.

### Expected Behavior

Server should start up and be accessible.

### To Reproduce

next.config.js
```javascript
const withTM = require('next-transpile-modules')(['@react-hook/latest', '@react-hook/debounce']); // pass the modules you would like to see transpiled

module.exports = withTM();

```

server.ts
```typescript
import * as dotenv from 'dotenv';
dotenv.config({
  path: './.env.local',
});
import { createServer } from 'http';
import next from 'next';
import { NextServer } from 'next/dist/server/next';

/*
    The Next docs include an example server.js file, but that shows how to override Next's routing with a custom implementation, 
    which we don't want to do because we have no requirement to do so 
    and because it disables some features such as Automatic Static Optimisation.

    We end up taking the source code for the Next server that ultimately runs when you run next start and then modified it to suit our needs. 
*/
interface ServerConfig {
  hostname: string;
  port: any;
  env: string;
}

const startServer = async (config: ServerConfig): Promise<NextServer> => {
  const serverOptions = {
    dev: config.env === 'development',
    dir: '.',
    quiet: false,
  };

  const app = next(serverOptions);
  const handleNextRequests = app.getRequestHandler();

  const srv = createServer((req, res) => {
    handleNextRequests(req, res);
  });

  await new Promise<void>((resolve, reject) => {
    // This code catches EADDRINUSE error if the port is already in use
    srv.on('error', reject);
    srv.on('listening', () => resolve());
    srv.listen(config.port, config.hostname);
  });

  // It's up to caller to run `app.prepare()`, so it can notify that the server
  // is listening before starting any intensive operations.
  return app;
};

const serverConfig: ServerConfig = {
  hostname: 'localhost',
  port: process.env.PORT || 3000,
  env: process.env.APP_ENV || process.env.NODE_ENV || 'production',
};

startServer(serverConfig)
  .then(async (app) => {
    console.log(
      `started server on host ${serverConfig.hostname}, port ${serverConfig.port}, env ${serverConfig.env}`
    );

    await app.prepare();
  })
  .catch((err) => {
    console.error(err);

    process.exit(1);
  });

```

**Comment by timneutkens** [comment]
This file no longer exists in the latest version. Please also make sure you don't import Next.js internals like `import { NextServer } from 'next/dist/server/next';`.

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
